File: Frontend/src/modules/posts/services/postService.js
import api from '@/services/api';

const postService = {
  getFeatured: async () => {
    const response = await api.get('/posts/featured');
    return response.data;
  },
  getPaged: async (pageIndex = 1, pageSize = 10, keyword = '', category = 'ALL', status = 'ALL') => {
    const response = await api.post('/posts/page', {
      pageIndex,
      pageSize,
      keyword,
      category,
      status,
    });
    return response.data;
  },
  getById: async (id) => {
    const response = await api.get(`/posts/id/${id}`);
    return response.data;
  },
  create: async (data) => {
    const response = await api.post('/posts', data);
    return response.data;
  },
  update: async (id, data) => {
    const response = await api.put(`/posts/${id}`, data);
    return response.data;
  },
  delete: async (id) => {
    const response = await api.delete(`/posts/${id}`);
    return response.data;
  }
};

export default postService;

File: Backend/src/main/java/com/family/modules/post/service/PostService.java
package com.family.modules.post.service;

import com.family.common.dto.PagingRequest;
import com.family.modules.post.dto.PostRequest;
import com.family.modules.post.entity.Post;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface PostService {
    Post getById(UUID id);
    Post getBySlug(String slug);
    List<Post> getFeaturedPosts();
    Page<Post> getPaged(PagingRequest request);
    Post create(PostRequest request, UUID authorId);
    Post update(UUID id, PostRequest request);
    void delete(UUID id);
}
